cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -143,11 +143,10 @@ private newtype TDefOrUseImpl =
 private predicate isGlobalUse(
   GlobalLikeVariable v, IRFunction f, int indirection, int indirectionIndex
 ) {
-  exists(VariableAddressInstruction vai, int defIndex |
+  exists(VariableAddressInstruction vai |
     vai.getEnclosingIRFunction() = f and
     vai.getAstVariable() = v and
-    isDef(_, _, _, vai, indirection, defIndex) and
-    indirectionIndex = [0 .. defIndex] + 1
+    isDef(_, _, _, vai, indirection, indirectionIndex)
```

Comment:
The change seems to offset the final argument of `isDef` by 1, is that correct?

Reply:
Correct. So after this change any write to a global variable with indirection index `indirectionIndex` generates a `GlobalUse` SSA read at the end of the enclosing function with indirection index `indirectionIndex`.